.github/workflows/zephyr-unit-tests.yaml

```diff
@@ -0,0 +1,46 @@
+name: Zephyr - Tests
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  cpu-unit-test:
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    strategy:
+      matrix:
+        python-version: ["3.12"]
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install uv
+        uses: astral-sh/setup-uv@v6
+        with:
+          version: "0.7.20"
+          enable-cache: true
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          uv sync --package zephyr --group test --frozen
+
+      - name: Test with pytest
+        env:
+          CI: true
```

Comment:
random: isn't this the default https://docs.github.com/en/actions/reference/workflows-and-actions/variables#default-environment-variables ?